shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -47,11 +43,51 @@
     // The texture registry maintaining the textures into which the embedded views will be rendered.
     private TextureRegistry mTextureRegistry;
 
-    // The messenger used to communicate with the framework over the platform views channel.
-    private BinaryMessenger mMessenger;
+    // The system channel used to communicate with Flutter about platform views.
+    private PlatformViewsChannel platformViewsChannel;
 
     private final HashMap<Integer, VirtualDisplayController> vdControllers;
 
+    private final PlatformViewsChannel.PlatformViewsHandler channelHandler = new PlatformViewsChannel.PlatformViewsHandler() {
```

Comment:
What's the benefit of having this anonymous class over making PlatformViewsController implement the interface?

Reply:
The distinction is based on the intended use. Implementing an interface is a public statement that the given instance can be used as the interface. In other words, a PlatformViewsController could be used by client code as a PlatformViewsHandler.  But if another area of code used this class as a PlatformViewsHandler, that would very likely break this code. What we really want is for this PlatformViewsController to own both a PlatformViewsChannel, as well as a PlatformViewsHandler. These are internal implementation details of a PlatformViewsController, rather than a public behavior of PlatformViewsController. Therefore, the PlatformViewsHandler is kept private and internal.

Reply:
I see.

What do you think about keeping the actual method implementations in this anonymous class (e.g inlining `PlatformViewsController#createPlatformView` here).

We're adding quite a lot of indirection layers here, I'm convinced that all of them are justified but this one(see my note to self in the PR comment).

Reply:
So the recommendation is to avoid re-declaring each method within the outer class and to simply do the behavior within the anonymous class?

If so, then I think that should be fine. I think the only reason I didn't do that was because I was trying to create a minimal diff against the existing code. I wasn't sure at the time if you were in the middle of evolving this class. 

So if you confirm that this change is a good resolution then I'll take care of it.

Reply:
Yeah I support inlining the implementation if there isn't a good reason to keep the methods separate.